src/v/test_utils/BUILD

```diff
@@ -82,3 +82,21 @@ redpanda_test_cc_library(
     ],
     visibility = ["//visibility:public"],
 )
+
+redpanda_test_cc_library(
+    name = "tmp_dir",
+    hdrs = [
+        "tmp_dir.h",
+    ],
+    include_prefix = "test_utils",
+    visibility = ["//visibility:public"],
+)
+
+redpanda_test_cc_library(
+    name = "async",
+    hdrs = [
+        "async.h",
```

Comment:
This is also exposed in `:gtest`.

Can we remove it from there?

Reply:
This was a bug on my part, I didn't see that seastar_boost already existed, which was the correct dep for the e2e test, changed in next push.